union-ai-admin/aws/script/generate.py

```diff
@@ -655,22 +741,19 @@ def create_admin_policy(role_type):
                         Action("iam", "PassRole"),
                     ],
                     Resource=[
-                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*"),
-                        Sub("arn:aws:iam::${AWS::AccountId}:role/*"),
-                    ],
-                ),
-                Statement(
-                    Effect=Allow,
-                    Action=[
-                        Action("autoscaling", "CreateOrUpdateTags"),
-                        Action("autoscaling", "DeleteTags"),
+                        Sub(
+                            "arn:aws:iam::${AWS::AccountId}:role/aws-service-role/*.amazonaws.com/*"
+                        ),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/opta-*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/unionai-*"),
                     ],
-                    Resource=["*"],
                 ),
                 Statement(
                     Effect=Allow,
                     Action=[
                         Action("ec2", "EnableEbsEncryptionByDefault"),
+                        Action("ec2", "ModifyEbsDefaultKmsKeyId"),
+                        Action("ec2", "DisableEbsEncryptionByDefault"),
                         Action("ec2", "GetEbsEncryptionByDefault"),
                     ],
                     Resource=["*"],
```

Comment:
We can set a condition on this for the region where we will be enabling this. See https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html